### PR TITLE
Quit message mining

### PR DESCRIPTION
**Pull request by yhaspel**
Screen Shot 2019-09-03 at 4 39 26 PM

### CODE REVIEW

File: app/App.js [hunk 1]
```diff
@@ -47,6 +48,7 @@ class App extends React.Component<Props, State> {
       <>
         <GlobalStyle />
         <Provider store={store}>
+          <QuitDialog />
```
Comment (IlyaVi) [inline app/App.js]: move it to Main and increase css dimensions (top, bottom, left, right) to cover the whole screen

File: app/components/common/QuitDialog.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+const Wrapper = styled.div`
+  position: fixed;
+  top: 0;
+  bottom: 0;
+  left: 0;
+  right: 0;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  background-color: ${smColors.white};
```
Comment (IlyaVi) [inline app/components/common/QuitDialog.js]: make it more transparent 0.2 opacity

File: app/components/common/QuitDialog.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      const isMining = miningStatus === nodeConsts.IN_SETUP || miningStatus === nodeConsts.IS_MINING;
+      isMining ? this.setState({ isVisible: true }) : this.handleQuit();
+    });
+  }
+
+  handleQuit = () => ipcRenderer.send(ipcConsts.QUIT_APP);
+
+  handleKeepInBackground = () => {
+    this.setState({ isVisible: false });
+    ipcRenderer.send(ipcConsts.KEEP_RUNNING_IN_BACKGROUND);
```
Comment (IlyaVi) [inline app/components/common/QuitDialog.js]: send notification from here, set timeout to 5 seconds

File: desktop/eventListners.js [hunk 4]
```diff
@@ -67,6 +68,21 @@ const subscribeToEventListeners = ({ mainWindow }) => {
     NodeManager.hardRefresh({ browserWindow: mainWindow });
   });
 
+  ipcMain.on(ipcConsts.QUIT_APP, () => {
```
Comment (IlyaVi) [inline desktop/eventListners.js]: this is special case, move listeners right after sending REQUEST_CLOSE event

File: app/App.js (hunk 1 above)
Comment (yhaspel) [reply app/App.js]: Ok on the dimensions, but if it's moved to Main it would need to be in Auth as well to enable quitting from there as well.

Comment (IlyaVi) [reply app/App.js]: When user is logged out there should be no notification cause there is no control of node

File: desktop/eventListners.js [hunk 5]
```diff
@@ -67,6 +68,8 @@ const subscribeToEventListeners = ({ mainWindow }) => {
     NodeManager.hardRefresh({ browserWindow: mainWindow });
   });
 
+  ipcMain.on(ipcConsts.KEEP_RUNNING_IN_BACKGROUND, () => mainWindow.hide());
```
Comment (IlyaVi) [inline desktop/eventListners.js]: move this to main.dev too

File: desktop/eventListners.js [hunk 6]
```diff
@@ -1,3 +1,4 @@
+import path from 'path';
```
Comment (IlyaVi) [inline desktop/eventListners.js]: remove

File: app/App.js (hunk 1 above)
Comment (yhaspel) [reply app/App.js]: True, but the event is intercepted at main.dev. That means we still need to listen to 'REQUEST_CLOSE' in auth in order to complete the app quit, no? On app close in main dev we're just sending 'REQUEST_CLOSE'.

Comment (IlyaVi) [reply app/App.js]: simply check location before changing visibility and in case of auth close immediately

**Pull request merged**